src/Diagnostics.RuntimeHost/Controllers/DeploymentController.cs

```diff
@@ -0,0 +1,162 @@
+﻿using Diagnostics.Logger;
+using Diagnostics.ModelsAndUtils.Models;
+using Diagnostics.ModelsAndUtils.Models.Storage;
+using Diagnostics.RuntimeHost.Services;
+using Diagnostics.RuntimeHost.Services.CacheService;
+using Diagnostics.RuntimeHost.Services.StorageService;
+using Diagnostics.RuntimeHost.Services.DevOpsClient;
+using Diagnostics.RuntimeHost.Utilities;
+using Diagnostics.Scripts;
+using Diagnostics.Scripts.CompilationService.Utilities;
+using Diagnostics.Scripts.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Diagnostics.RuntimeHost.Controllers
+{
+    [Authorize]
+    [Produces("application/json")]
+    [Route("api/deploy")]
+    public class DeploymentController : Controller
+    {
+
+        private IRepoClient devopsClient;
+        protected ICompilerHostClient _compilerHostClient;
+        protected IStorageService storageService;
+        private IInvokerCacheService detectorCache;
+
+        public DeploymentController(IServiceProvider services, IConfiguration configuration, IInvokerCacheService invokerCache)
+        {
+            this.storageService = (IStorageService)services.GetService(typeof(IStorageService));
+            this.devopsClient = (IRepoClient)services.GetService(typeof(IRepoClient));
+            this._compilerHostClient = (ICompilerHostClient)services.GetService(typeof(ICompilerHostClient));           
+            this.detectorCache = invokerCache;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Post([FromBody] DeploymentParameters deploymentParameters)
+        {
+            // If all required parameters are empty, reject the request.
+            if(string.IsNullOrWhiteSpace(deploymentParameters.CommitId) && string.IsNullOrWhiteSpace(deploymentParameters.FromCommitId)
+                && string.IsNullOrWhiteSpace(deploymentParameters.ToCommitId) && string.IsNullOrWhiteSpace(deploymentParameters.StartDate)
+                && string.IsNullOrWhiteSpace(deploymentParameters.EndDate)) {
+                return BadRequest("The given deployment parameters are invalid");
+            }
+                     
+            DeploymentResponse response = new DeploymentResponse();
+            response.DeploymentGuid = Guid.NewGuid().ToString();
+            response.DeployedDetectors = new List<string>();
+            response.FailedDetectors = new Dictionary<string, string>();
+
+            var commitId = deploymentParameters.CommitId;
+            var timeTakenStopWatch = new Stopwatch();
+            timeTakenStopWatch.Start();
+            DiagnosticsETWProvider.Instance.LogRuntimeHostMessage($"Starting deployment operation for {response.DeploymentGuid}");
```

Comment:
@rekhaswaminathan maybe we should create new events for this since this is a new and pretty big feature.

Reply:
ya, I agree. Good to create a bunch of events specific to deployment. Easy to manage and alert on